### PR TITLE
Add tools version 5.6 and set it as the current version for `main` branch

### PR DESCRIPTION
**Pull request by abertelrud**
Add tools version 5.6 and set it as the current version for `main` branch

### Motivation:

It's not clear yet whether this will be the next version, but we need a new version in order to remove the gates for the plugin support.  We can always adjust it later when the next version of Swift gets an official version number.

### Modifications:

- add `v5_6` enum
- set 5.6.0 as the current SwiftPM version (but keep it marked as being in development)

A separate PR will adjust the plugin availability.

### CODE REVIEW

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by WowbaggersLiquidLunch** [comment]
> It's not clear yet whether this will be the next version

It seems that apple/swift#38574 suggests that 5.6 is the next version.

**Comment by abertelrud** [comment]
> > It's not clear yet whether this will be the next version
> 
> It seems that [apple/swift#38574](https://github.com/apple/swift/pull/38574) suggests that 5.6 is the next version.

Ah, well spotted!  Thanks!  I usually wait for an announcement but in this case I wanted to move forward with the plugins.

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
@abertelrud good to merge? I would like to rebase #3635 on top when it is

**Comment by abertelrud** [comment]
Good to merge — it was just waiting for CI and approval.  Merging now.

**Pull request merged**
